src/main/java/seedu/weme/logic/LogicManager.java

```diff
@@ -26,19 +26,19 @@
 
     private final Model model;
     private final Storage storage;
-    private final MemeBookParser memeBookParser;
 
     public LogicManager(Model model, Storage storage) {
         this.model = model;
         this.storage = storage;
-        memeBookParser = new MemeBookParser();
     }
 
     @Override
     public CommandResult execute(String commandText) throws CommandException, ParseException {
         logger.info("----------------[USER COMMAND][" + commandText + "]");
 
         CommandResult commandResult;
+
+        MemeBookParser memeBookParser = model.getContext().getParser();
```

Comment:
Isn't this violation of Law of Demeter?

Reply:
Should we follow it? Context in this case is just that Logic can see ModelContext, which is not unusual in this case.

Reply:
I just feel like we should instantiate it somewhere. Not necessary but it feels odd. ImagePath has such a chaining method as well.

Reply:
My take is that `ModelContext` should not depend on the `Parser`s. What about
```
MemeBookParser memeBookParser = MemeBookParser.forContext(model.getContext());
```
?

Reply:
I support that. Parser class should generate parser based on context since it has to know the context. The converse is not necessary (context needs not to hold a parser)

Reply:
So this means the Parser will read the current model's context and return the appropriate parser to use to parse the command. Sounds good to me.